### PR TITLE
Weapons models update, optimization & cleanup

### PR DESCRIPTION
**Pull request by Desoroxxx**
## 🤔 What type of PR is this? (check all applicable)

- [x] 🍕 Addition
- [x] ⌨️ Productivity
- [x] 🐛 Bug Fix
- [x] 🔥 Optimization
- [ ] ⚙️ Configuration
- [ ] 🌟 Quality Of Life
- [x] ✨ Enhancement
- [ ] 📝 Documentation

## 📝 Description

TBD

## 🧐 The Rationale

TBD

## 🎯 Key Objectives

TBD

## 🚦 Testing 

TBD

## ⏮️ Backwards Compatibility 

This will make weapons in saves disappear, but doesn't matter since we are not even in beta yet.

## 📚 Related Issues & Documents

TBD

## 🖼️ Screenshots/Recordings

TBD

## 📖 Added to documentation?

- [ ] 📜 README.md
- [ ] 📓 Javadoc
- [ ] 🍕 Comments
- [x] 🙅 no documentation needed

### CODE REVIEW

**Comment by Braiam23** [comment]
ready for review

**Comment by Desoroxxx** [comment]
Fixes #284

**Comment by Koud-Wind** [comment]
@Desoroxxx Please check these factory files and their aiming positions

File: src/main/java/com/paneedah/mwc/items/guns/M79Factory.java [hunk 1]
```diff
@@ -1,612 +1,173 @@
...
 
-import java.util.Arrays;
-
 public class M79Factory implements GunFactory {
 
     public Item createGun(CommonProxy commonProxy) {
         return new Weapon.Builder()
+                .withName("m79")
+                .withAmmoCapacity(1)
+                .withFireRate(0.1f)
```
Comment (Paneedah) [inline src/main/java/com/paneedah/mwc/items/guns/M79Factory.java]: No balance changes until 0.2.
```suggestion
                .withFireRate(0.7f)
```

File: src/main/java/com/paneedah/mwc/items/guns/M79Factory.java [hunk 2]
```diff
@@ -1,612 +1,173 @@
...
+                                    new Transform()
+                                            .withPosition(-0.320000f, 0.140000f, 2.040000f)
+                                            .withBBRotation(10.0931F, 10.9576F, -10.0294F)
+                                            .withScale(3, 3, 3.5F)
+                                            .withPivotPoint(0, 0, 0)
+                                            .applyTransformations();
+                                })
+
+                        .build())
+                .withSpawnEntityDamage(40f)
```
Comment (Paneedah) [inline src/main/java/com/paneedah/mwc/items/guns/M79Factory.java]: No balance changes until 0.2.
```suggestion
                .withSpawnEntityDamage(70f)
```

File: src/main/java/com/paneedah/mwc/items/guns/M9A1Factory.java [hunk 3]
```diff
@@ -21,588 +17,489 @@
...
 
-import java.util.Arrays;
-
 public class M9A1Factory implements GunFactory {
 
     public Item createGun(CommonProxy commonProxy) {
         return new Weapon.Builder()
+                .withName("m9a1")
+                .withFireRate(0.5f)
+                .withRecoil(5f)
```
Comment (Paneedah) [inline src/main/java/com/paneedah/mwc/items/guns/M9A1Factory.java]: No balance changes until 0.2.
```suggestion
                .withRecoil(2f)
```

File: src/main/java/com/paneedah/mwc/items/guns/M9A1Factory.java [hunk 4]
```diff
@@ -21,588 +17,489 @@
...
+                .withReloadSound("m9a1_reload")
+                .withUnloadSound("pistol_unload")
+                .withInspectSound("inspection")
+                .withDrawSound("handgun_draw")
+                .withReloadingTime(40)
+                .withFlashIntensity(0.5f)
+                .withFlashScale(() -> 0.6f)
+                .withFlashOffsetX(() -> 0.13f)
+                .withFlashOffsetY(() -> 0.14f)
+                .withInaccuracy(2)
```
Comment (Paneedah) [inline src/main/java/com/paneedah/mwc/items/guns/M9A1Factory.java]: No balance changes until 0.2.
```suggestion
                .withInaccuracy(3)
```

File: src/main/java/com/paneedah/mwc/items/guns/MAC10Factory.java [hunk 5]
```diff
@@ -21,721 +16,591 @@
...
+                .withReloadSound("mac10_reload")
+                .withUnloadSound("mac10_unload")
+                .withInspectSound("inspection")
+                .withDrawSound("noaction_draw")
+                .withReloadingTime(45)
+                .withFlashIntensity(0.5f)
+                .withFlashScale(() -> 0.6f)
+                .withFlashOffsetX(() -> 0.16f)
+                .withFlashOffsetY(() -> 0.16f)
+                .withInaccuracy(2f)
```
Comment (Paneedah) [inline src/main/java/com/paneedah/mwc/items/guns/MAC10Factory.java]: No balance changes until 0.2.
```suggestion
                .withInaccuracy(3f)
```

File: src/main/java/com/paneedah/mwc/items/guns/MP40Factory.java [hunk 6]
```diff
@@ -3,1517 +3,159 @@
...
 import org.lwjgl.opengl.GL11;
 
-import java.util.Arrays;
-
 public class MP40Factory {
 
     public Item createGun(CommonProxy commonProxy) {
         return new Weapon.Builder()
+                .withName("mp40")
+                .withFireRate(0.6f)
```
Comment (Paneedah) [inline src/main/java/com/paneedah/mwc/items/guns/MP40Factory.java]: No balance changes until 0.2.
```suggestion
                .withFireRate(0.55f)
```

File: src/main/java/com/paneedah/mwc/items/guns/MP5A5Factory.java [hunk 7]
```diff
@@ -21,875 +16,731 @@
...
 
-import java.util.Arrays;
-
 public class MP5A5Factory {
 
     public Item createGun(CommonProxy commonProxy) {
         return new Weapon.Builder()
+                .withName("mp5a5")
+                .withFireRate(0.8f)
+                .withRecoil(1.7f)
```
Comment (Paneedah) [inline src/main/java/com/paneedah/mwc/items/guns/MP5A5Factory.java]: No balance changes until 0.2.
```suggestion
                .withRecoil(1.5f)
```

**Review by Paneedah — CHANGES_REQUESTED** [review]
This is my first review. I have to go for now.

File: src/main/java/com/paneedah/mwc/items/guns/ACRFactory.java [hunk 8]
```diff
@@ -20,1072 +15,912 @@
...
+                .withFireRate(0.65f)
+                .withRecoil(3f)
+                .withZoom(0.9f)
+                .withConfigGroup(GunConfigurationGroup.ASSAULT_RIFLE)
+                .withMuzzlePosition(new Vec3d(-0.1, -1.14, -6.22))
+                .hasFlashPedals()
+                .withMaxShots(1, 3, Integer.MAX_VALUE)
+                .withShootSound("acr")
+                .withSilencedShootSound("m4a1_silenced")
+                .withDrawSound("acr_draw")
```
[the remark this answers is not in the document]
Comment (Braiam23) [reply src/main/java/com/paneedah/mwc/items/guns/ACRFactory.java:32]: I will not add something that is unnecessary

File: src/main/java/com/paneedah/mwc/items/guns/AACHoneyBadgerFactory.java [hunk 9]
```diff
@@ -1001,7 +959,7 @@ else if(model instanceof AcogReticle) {
                                 }, 250, 0))
 
                         .build())
-                .withSpawnEntityDamage(5.4f)
+                .withSpawnEntityDamage(6.4f)
```
[the remark this answers is not in the document]
Comment (Braiam23) [reply src/main/java/com/paneedah/mwc/items/guns/AACHoneyBadgerFactory.java:962]: 👍

File: src/main/java/com/paneedah/mwc/items/guns/APC9Factory.java [hunk 10]
```diff
@@ -21,53 +19,41 @@
...
-                .withUnloadSound("mpx_unload")
-                .withInspectSound("inspection")
                 .withDrawSound("noaction_draw")
                 .withReloadingTime(50)
                 .withFlashIntensity(0.5f)
                 .withFlashScale(() -> 0.6f)
                 .withFlashOffsetX(() -> 0.12f)
                 .withFlashOffsetY(() -> 0.15f)
-                .withInaccuracy(1.1f)
+                .withInaccuracy(1f)
```
[the remark this answers is not in the document]
Comment (Braiam23) [reply src/main/java/com/paneedah/mwc/items/guns/APC9Factory.java:40]: No

File: src/main/java/com/paneedah/mwc/items/guns/ACRFactory.java (hunk 8 above)
[the remark this answers is not in the document]
Comment (Koud-Wind) [reply src/main/java/com/paneedah/mwc/items/guns/ACRFactory.java:32]: > Re-add the removed sounds. Unsure why this was removed in the first place.

Please wait for me to restore it, btw I would like to know the reason

File: src/main/java/com/paneedah/mwc/items/guns/APC9Factory.java (hunk 10 above)
[the remark this answers is not in the document]
Comment (Koud-Wind) [reply src/main/java/com/paneedah/mwc/items/guns/APC9Factory.java:40]: > No balance changes until 0.2 please.

Please wait for me to restore it, btw I would like to know the reason

**Review by Paneedah — APPROVED** [review]

**Pull request merged**